### ISSUE TITLE
Eclipse Planning 2024 - Hero 10 & 11

### ISSUE THREAD

**Issue by jrsbww1** [gopro/labs#835]
I have a Hero 10 and an 11....Installed the appropriate firmware for each.   No issues there.   

Did several tested using the eclipse time-lapse QR code....and didn't get expected results.   Assuming my expectations are correct.   I did several tests... and eventually got them both set the same for a side by side test.  

The cameras started on time. But neither executed as expected.   

The 11 stopped recording at the totality start time.   I would have expected it to start new recording at 2 minutes before totality.  3 minutes later it powered down. 

The 10 went further, but did not complete as expected.  Again at totality start time it stopped recording 1 and started recording 2 several seconds later.   So again it started that recording 2 min later than expected and ended the second recording at the time I expected.  But never did the third recording.    Recording 1 had frames every 10 seconds, that's correct.   Recording two was supposed to be every 1/2 second... as my digital clock was only showing seconds, I saw 1 sec increments, I suspect it was correct.  But there was not a 3rd recording of 10 second frames. 

This is a really cool idea.  I like the QR code settings... I just don't use my cameras enough.  This would have been the ideal time for something like this.   But my weather forecast keeps looking worse...  so probability I need this is slim... cutting my loses and punting on this little project.   

I didn't see a lot of comments... hopefully this helps someone...   Being an IT person... this was right up my alley...

**Comment by dnewman-gpsw** [comment]
Thank you for your report.  I expect this is just getting the correct settings.  There are four different options, which did you test?  Only option 1 will certainly work on HERO10, the other options are only maybes for that camera (I did not test them.)  HERO11 (and 12) should be fine with the latest Labs firmware on both.

Please share the command you ran, this part:
![image](https://github.com/gopro/labs/assets/10856084/a847c7c2-f956-4931-8979-db3f55dc72b7)
 
or just press "Copy URL to Clipboard" and paste that like this:
```https://gopro.github.io/labs/control/set/?cmd=oT240402135941"Eclipse TL
Type2"!11:00NmTp.10tb1w55i1M1sMoMEXPX=30!S!13:04EoMEXPX=0&title=Eclipse%20Timelapse```

 > The 11 stopped recording at the totality start time. I would have expected it to start new recording at 2 minutes before totality. 3 minutes later it powered down.

None of the commands are set to behave that way.   Types 1, 2 & 3, only capture one clip (night or time-lapse), starting well before and ending well after totality. There is no change at totality.   

Type 4 only captures two clips, first starting an hour before (default config), switching at totality and ending 2 minutes after totality ends.   This is the most risky, and with two cameras, there is no need for the added risk, as one camera can do each type that type 4 represents.

If this is a problem with have 5+ days to fix it.

**Comment by dnewman-gpsw** [comment]
More details please, I want to see if we have a bug in the scripts.

**Comment by jrsbww1** [comment]
Opps... I wrote a reply last night... I think I didn't click on comment to send it... Opps.... I apologize... 

Is it intentional that when I scan a barcode, that the Wifi is disabled?  I did another test specific for this... wifi is set to on... when I scan the code and have a start time... wifi is definitely turned off.   It does save battery, but I would like to connect to get a status.  I'd rather not have to stick my face in the camera to see if it's still recording.   

I still had my code up..   It was Type 4.... 
GoProQR: oT240402171706"Eclipse TL Type4"!15:35NmNLp.10tb1w55i1M1sMoMEXPX=30!S!598EpeAi8M1sMoMEXPX=1!S!15:49EoMEXPX=0
Eclipse Timelapse

![Screenshot 2024-04-02 at 3 28 46 PM](https://github.com/gopro/labs/assets/4573224/8c526059-e102-4cf9-a02d-386711fa636a)

Hmmm, but task 2 & 3 both say  "before and after totality", that's what made me expect 3 videos.  Job is only partly done for the exit isn't captured.  I didn't get any after totality video... for a type 4 I would expect 3 videos... and the timing of video 2 was wrong... it started at totality... not 2 minutes before, but it did go until 2 minutes after totality

![Screenshot 2024-04-03 at 8 28 59 PM](https://github.com/gopro/labs/assets/4573224/2fca78c5-a544-4b2e-beb9-1b8baba556ef)
This second screen shot is a table I made... times, what I expected (I could have incorrect expe3ctations) what Hero10
did and what hero 11 did.

No Worries, I'm just going to set them manually.   I set them last night to just start a c1+2min and end at c-2min.   they both ran at the expected time and duration. I still need to check the video though

**Comment by dnewman-gpsw** [comment]
I expect the racing through commands, trying to switch recording modes quickly, is it is not always working.  I have added ```!1N``` for a one second delay to improve the robustness. Otherwise it seems to be working.   Already updated https://gopro.github.io/labs/control/eclipse/

Thank you this detailed report.

The WiFi is turned off for any sleep event, but you can have the script on it back on gain.   In this example an ```oW1``` after the ```!15:35N```, so as soon as it wakes the WiFi is turned on.

**Comment by Jahnkeanater** [comment]
I can confirm that this failed miserably on a hero 11 black. Trying to record at 5.3k 4:3. The camera turned on and if you tapped the touch screen it showed you a preview with no UI. But it never recorded anything. I quickly generated another QR code to give it another chance to start recording but after 98 seconds it did the exact same thing. I ended up just manually starting a nightlapse right before totality. It was a huge headache to have to screw around with the camera that was supposed to be completely automatic.

**Comment by dnewman-gpsw** [comment]
Sorry you had issues. Sounds like you're had one of the cameras that needed a factory reset after the new firmware is installed. We'd hope that all of these sort of glitches would have been found in prior day testing.  Yes, when working, it's fully automatic.

**Comment by jrsbww1** [comment]
Both my 10 & 11 fired up and executed fine.   The results were OK, but
partly due to heavy overcast skies... great time lapse of clouds fly by.
 I notice that while the cameras were set differently the end results video
was very much the same.  THe dark of totality was about 1-2 seconds, while
I was hoping to get longer totality with teh settings on one.  But this all
may be because of the cloud cover.  I need to think about that.

I used the app on my ipad to program them.   Though I found a few things
not right.   THe app kept losing the lens setting... I had to
constantly set it back to Wide.    I had to use Set date/time otherwise the
11 time was overwritten with some garbage time from the barcode.  Even then
the time would freeze causing me to restart the app.  It was unnerving not
seeing anything on the screen when the camera fired up.  For aligning the
static shot, I had to do that, then show the camera the barcode on the
ipad... rather awkward.  It was only after many, many test sessions that I
had worked through that I was reasonably sure the camera would fire
correctly.   I did not use mode 4 at all.

I love the idea of the bar code app... would really be nice to save a
config file, with a field for comments so I could make notes in each setup
file.  Understand it's all Beta... was cool to play with.   On my way to
Florida for a couple weeks... planning to play some more while watching
sunsets :-)
